osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -253,46 +253,35 @@ public bool SelectBeatmap(BeatmapInfo beatmap, bool bypassFilters = true)
         /// <param name="skipDifficulties">Whether to skip individual difficulties and only increment over full groups.</param>
         public void SelectNext(int direction = 1, bool skipDifficulties = true)
         {
-            var visibleItems = Items.Where(s => !s.Item.Filtered.Value).ToList();
-
-            if (!visibleItems.Any())
-                return;
-
-            DrawableCarouselItem drawable = null;
-
-            if (selectedBeatmap != null && (drawable = selectedBeatmap.Drawables.FirstOrDefault()) == null)
-                // if the selected beatmap isn't present yet, we can't correctly change selection.
-                // we can fix this by changing this method to not reference drawables / Items in the first place.
+            if (!beatmapSets.Where(s => !s.Filtered.Value).ToList().Any())
```

Comment:
```suggestion
            if (beatmapSets.All(s => !s.Filtered.Value))
```
The `ToList()` is gratuitous and adds an overhead of an additional list allocation.

Reply:
You forgot to invert the logic :/

Reply:
that I did. resharper fixed it up for me automatically in IDE but I missed it when copying over here